File: backend/src/main/java/com/toth/akos/nexused/enums/Role.java
package com.toth.akos.nexused.enums;

import org.springframework.security.core.GrantedAuthority;

public enum Role implements GrantedAuthority {
    ADMIN,
    STUDENT,
    PARENT,
    TEACHER;

    @Override
    public String getAuthority() {
        return this.name();
    }
}

File: frontend/src/app/web/common/util/enums/Role.ts
export enum Role {
    ADMIN = 'ADMIN',
    STUDENT = 'STUDENT',
    PARENT = 'PARENT',
    TEACHER = 'TEACHER'
}

export const ROLE_TYPE = {
    [Role.ADMIN]: 'Admin',
    [Role.STUDENT]: 'Tanuló',
    [Role.PARENT]: 'Szülő',
    [Role.TEACHER]: 'Tanár'
};

export function getRoleName(value: Role | null) {
    return value ? ROLE_TYPE[value] : '';
}
